pkg/distributor/distributor.go

```diff
@@ -808,6 +820,36 @@ func (d *Distributor) checkUsageGroupsIngestLimit(tenantID string, groupsInReque
 	return nil
 }
 
+func (d *Distributor) shouldSample(tenantID string, groupsInRequest []string) bool {
+	l := d.limits.SamplingProbability(tenantID)
+	if l == nil {
+		return true
+	}
+
+	// Determine the minimum probability among all matching usage groups.
+	minProb := 1.0
```

Comment:
Wonder if it wouldn't be better with the `maximum` of those, you could have a very coarse background profiling `0.01` rule for all services and the profiles you are interested in (with service_name/pods selector) would be overriding them with out to 1.0.